modules/typeable/src/main/scala/shapeless3/typeable/typeable.scala

```diff
@@ -91,8 +91,32 @@ object Typeable extends Typeable0 {
   /** Typeable instance for `Unit`. */
   given unitTypeable: Typeable[Unit] = ValueTypeable[Unit, scala.runtime.BoxedUnit](classOf[scala.runtime.BoxedUnit], "Unit")
 
+  /** Typeable instance for `java.lang.Byte`. */
+  given jlByteTypeable: Typeable[jl.Byte] = ValueTypeable[jl.Byte, jl.Byte](classOf[jl.Byte], "java.lang.Byte")
+  /** Typeable instance for `java.lang.Short`. */
+  given jlShortTypeable: Typeable[jl.Short] = ValueTypeable[jl.Short, jl.Short](classOf[jl.Short], "java.lang.Short")
+  /** Typeable instance for `java.lang.Character`. */
+  given jlCharacterTypeable: Typeable[jl.Character] = ValueTypeable[jl.Character, jl.Character](classOf[jl.Character], "java.lang.Character")
+  /** Typeable instance for `java.lang.Integer`. */
+  given jlIntegerTypeable: Typeable[jl.Integer] = ValueTypeable[jl.Integer, jl.Integer](classOf[jl.Integer], "java.lang.Integer")
+  /** Typeable instance for `java.lang.Long`. */
+  given jlLongTypeable: Typeable[jl.Long] = ValueTypeable[jl.Long, jl.Long](classOf[jl.Long], "java.lang.Long")
+  /** Typeable instance for `java.lang.Float`. */
+  given jlFloatTypeable: Typeable[jl.Float] = ValueTypeable[jl.Float, jl.Float](classOf[jl.Float], "java.lang.Float")
+  /** Typeable instance for `java.lang.Double`. */
+  given jlDoubleTypeable: Typeable[jl.Double] = ValueTypeable[jl.Double, jl.Double](classOf[jl.Double], "java.lang.Double")
+  /** Typeable instance for `java.lang.Boolean`. */
+  given jlBooleanTypeable: Typeable[jl.Boolean] = ValueTypeable[jl.Boolean, jl.Boolean](classOf[jl.Boolean], "java.lang.Boolean")
+  /** Typeable instance for `scala.runtime.BoxedUnit`. */
+  given srBoxedUnitTypeable: Typeable[scala.runtime.BoxedUnit] = ValueTypeable[scala.runtime.BoxedUnit, scala.runtime.BoxedUnit](classOf[scala.runtime.BoxedUnit], "scala.runtime.BoxedUnit")
+
   def isAnyValClass[T](clazz: Class[T]) =
-    (classOf[jl.Number] isAssignableFrom clazz) ||
+    clazz == classOf[jl.Byte] ||
+    clazz == classOf[jl.Short] ||
+    clazz == classOf[jl.Integer] ||
+    clazz == classOf[jl.Long] ||
+    clazz == classOf[jl.Float] ||
+    clazz == classOf[jl.Double] ||
```

Comment:
Pretty sure that was fixed in shapeless 2 but maybe after it was ported

Reply:
I think I undid on the shapeless-3 branch because there was on sjs support, so it wasn't needed.